safety_shield/tests/src/robot_reach_tests.cc

```diff
@@ -1,43 +1,151 @@
 #include <gtest/gtest.h>
-#include <math.h> 
+#include <math.h>
 
 #include <Eigen/Dense>
-#include "spdlog/spdlog.h" 
 
 #include "robot_reach_fixture.h"
 #include "safety_shield/robot_reach.h"
+#include "spdlog/spdlog.h"
 
 namespace safety_shield {
 
-TEST_F(RobotReachTest, InitializationTest){
+TEST_F(RobotReachTest, InitializationTest) {
   EXPECT_DOUBLE_EQ(0, 0.0);
 }
 
-TEST_F(RobotReachTest, ForwardKinematicTest0){
+TEST_F(RobotReachTest, ForwardKinematicTest0) {
   Eigen::Matrix4d result = Eigen::Matrix4d::Identity();
   robot_reach_->forwardKinematic(0.0, 0.0, result);
   Eigen::Matrix4d expect;
-  expect <<   1, 0, 0, 0, 
-              0, 0, -1, 0, 
-              0, 1, 0, 0.1, 
-              0, 0, 0, 1;
+  expect << 1, 0, 0, 0, 0, 0, -1, 0, 0, 1, 0, 0.1, 0, 0, 0, 1;
   EXPECT_TRUE(result.isApprox(expect));
 }
 
-TEST_F(RobotReachTest, ForwardKinematicTest1){
+TEST_F(RobotReachTest, ForwardKinematicTest1) {
   Eigen::Matrix4d result = Eigen::Matrix4d::Identity();
-  robot_reach_->forwardKinematic(M_PI/2, 0.0, result);
+  robot_reach_->forwardKinematic(M_PI / 2, 0.0, result);
   Eigen::Matrix4d expect;
-  expect << 0.0, -1.0, 0.0, 0.0, 
-            0.0, 0.0, -1.0, 0.0, 
-            1.0, 0.0, 0.0, 0.1, 
-            0.0, 0.0, 0.0, 1.0;
+  expect << 0.0, -1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 1.0, 0.0, 0.0, 0.1, 0.0, 0.0, 0.0, 1.0;
   EXPECT_TRUE(result.isApprox(expect));
 }
 
-} // namespace safety_shield
+/*
 
-int main(int argc, char **argv){
-    testing::InitGoogleTest(&argc, argv);
-    return RUN_ALL_TESTS();
+TEST_F(RobotReachTestVelocity, JacobianTest0) {
```

Comment:
we can delete the (commented out) tests, because the capsule jacobians are different from the ones calculated with pinocchio

Reply:
Could you write some tests for the Jacobian calculation with a dummy robot? E.g., with a 2-link planar robot as defined in Siciliano_2009_Robotics Modelling, Planning and Control?
This can be in another PR.